### ISSUE TITLE
Cellmax is not saved in the restart file if changed by user

### ISSUE THREAD

**Issue by aklothakis** [sparta/sparta#162]
## Summary

Cellmax variable is not saved and cannot be read by the read_restart function if user changes it.

## Type of Issue

Bug Report

## SPARTA Version (Bug Report)

Latest version downloaded from Github

## Expected Behavior (Bug Report)

The code should be able to restart if the user sets the global cellmax variable to a value different than 100. 

## Actual Behavior (Bug Report)

When you try to restart the code with a restart script when it reads the restart file the it pops an error message saying that the global cellmax is not set although it is set in the input script.

I checked the read_restart, write_restart and i saw that the CELLMAX value is not taken into account. Tried to add to enumerate the CELLMAX changing some other lines (in order to write and read the new variable) but then i get error messages from the processes.

## Steps to Reproduce (Bug Report)
Make a surface file with more than 100 surfaces per cell and then set the global cellmax variable to something else than 100. When you try to restart you get an error message that the cellmax value is not set.

**Comment by aklothakis** [comment]
When i change the value of maxcellpersurf manually in the grid.cpp file the code runs and restarts fine but when set in the input script it doesn't

**Comment by stanmoore1** [comment]
I can't seem to reproduce this. Are you sure your `global cellmax` command is *before* the `read_restart` command?

**Comment by stanmoore1** [comment]
[in.sphere_start.txt](https://github.com/sparta/sparta/files/4787301/in.sphere_start.txt)
[in.sphere_restart.txt](https://github.com/sparta/sparta/files/4787302/in.sphere_restart.txt)

Here are the two input files I used to test. If I don't set the cellmax then I get this error:

```
Max cells overlapping any surf = 228
ERROR: Too many cells overlap one surf - set global cellmax (../grid_surf.cpp:284)
```

So I used `global cellmax 300`.

**Comment by stanmoore1** [comment]
Just to clarify, the `global cellmax` command isn't stored by the restart file, so it has to be set again, before the `read_restart` command.

**Comment by stanmoore1** [comment]
@aklothakis I'm going to close this issue since I can't reproduce it, but please let us know if you have more info on this.

**Comment by aklothakis** [comment]
Hello there and thank you for your fast response. In the Manual page 321 says that the read_restart command stores global settings. If you check the read_restart.cpp and write_restart.cpp code you can see that indeed the global settings are saved apart from the cellmax setting.

In order to be able to reproduce the issue try restarting the code without the global command (global commands shouldn't be in the restart script because are saved in the restart file) in your restart script. You will see that all the global settings are saved but you will get a message that the cellmax value is not set. But the cellmax value is set in your input script.

This means that the cellmax value is not saved by the write_restart nor read by the read_restart.

**Comment by stanmoore1** [comment]
OK I see what you are saying now. The `surfmax` variable is stored in the restart file but `cellmax` is not. There are probably other global commands not included as well, like `mem/limit` and `particle/reorder`.

**Comment by aklothakis** [comment]
Ok i see... I will have a look at them too. In the pool request i have changed the appropriate lines in the code to include the cellmax. Tested it for a several timesteps and it seems to be working. I haven't tested it extensively though.

If you have any specific tests in mind let me know and i will perform them.


I have attached the "modified" restart to reproduce the issue.

Thanks again.


[in.sphere_restart_v2.txt](https://github.com/sparta/sparta/files/4788352/in.sphere_restart_v2.txt)

**Comment by stanmoore1** [comment]
Here are the global commands that are missing from the restart file:

- surfs
- surfgrid
- cellmax
- splitmax
- surftally
- surfpush
- particle/reorder
- mem/limit

**Comment by aklothakis** [comment]
Thank you i will have have a look at them and i will submit a pull request when i have something ready. 
Meanwhile tested the cellmax for my case for several thousand timesteps now with 200 cores and is still working.....

**Comment by stanmoore1** [comment]
Thanks for reporting and working on this. I can help out as well if needed.

**Comment by stanmoore1** [comment]
> Here are the global commands that are missing from the restart file:
> 
> * surfs
> * surfgrid
> * cellmax
> * splitmax
> * surftally
> * surfpush
> * particle/reorder
> * mem/limit

We are planning to fix this in #303.
